### PR TITLE
GA-20 Add new endpoint to retrieve Organisation Profile

### PR DESCRIPTION
**Pull request by helenbirdhmcts**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [x] the `keep-helm` label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/GA-20

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesRoleResult.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel(description = "role")
+public class AccessTypeRolesRoleResult {
+
+    private String caseTypeId;
+    private String organisationalRoleName;
+    private String groupRoleName;
+    private String caseGroupIdTemplate;
+
```
Comment (abul-syed) [inline repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesRoleResult.java:14]:
```suggestion
private String groupAccessEnabled;
```

I think we may also need `groupAccessEnabled` (not specified in Jira though)
but HLD suggests it's used in PRM to decide whether to create a ORG or CASE role

File: rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    @PostMapping(value = "/retrieve-access-types",consumes = {"application/json"}, produces = {"application/json"})
+    @ApiOperation(value = "Get all accessTypes for all caseTypes", response = AccessTypeRolesJurisdictionResult.class,
+        responseContainer = "List")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "List of valid access types request"),
+        @ApiResponse(code = 401, message = "Unauthorised request"),
+        @ApiResponse(code = 403, message = "Bad request")
+    })
+    public AccessTypeRolesJurisdictionResults retrieveAccessTypeRoles(
+        @Valid @RequestBody() @NotNull  OrganisationProfileIds organisationProfileIds) {
```
Comment (abul-syed) [inline rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java:57]:
```suggestion
        @Valid @RequestBody @NotNull OrganisationProfileIds organisationProfileIds) {
```

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesJurisdictionResult.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    private String id = null;
+    private String name = null;
+    private List<AccessTypeRolesResult> accessTypeResults = new ArrayList<AccessTypeRolesResult>();
+
+    /**
+     * id of a particular Jurisdiction.
+     **/
+    @ApiModelProperty(required = true, value = "")
+    @JsonProperty("jurisdictionid")
```
Comment (abul-syed) [inline repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesJurisdictionResult.java:21]:
```suggestion
    @JsonProperty("jurisdictionId")
```

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesJurisdictionResult.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@ApiModel(description = "jurisdiction")
+public class AccessTypeRolesJurisdictionResult {
+
+    private String id = null;
```
Comment (abul-syed) [inline repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesJurisdictionResult.java:13]: might be better to name these fields `jurisdictionId` & `jurisdictionName`, that way no need to use `@JsonProperty("jurisdictionId")` below

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesJurisdictionResults.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@ApiModel(description = "jurisdictions")
+public class AccessTypeRolesJurisdictionResults {
+
+    private List<AccessTypeRolesJurisdictionResult> jurisdictions = new ArrayList<AccessTypeRolesJurisdictionResult>();
```
Comment (abul-syed) [inline repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesJurisdictionResults.java:13]:
```suggestion
    private List<AccessTypeRolesJurisdictionResult> jurisdictions = new ArrayList<>();
```

File: rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+        List<AccessTypeRolesField> accessTypeRoles = null;
+
+        List<AccessTypeRolesEntity>  accessTypeRolesEntities = accessTypeRolesRepository
+            .findByOrganisationProfileIds(organisationProfileIds.getOrganisationProfileIds());
+        accessTypeRoles =  accessTypeRolesEntities.stream().map(entityToResponseDTOMapper::map)
+            .collect(Collectors.toList());
+
```
Comment (abul-syed) [inline rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java:64]: you also need to add case where organisationProfileIds is not sent - for this scenario all access type defs need to be returned. required for PRM batch process

File: rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+        List<AccessTypeRolesJurisdictionResult> jurisdictions = buildJuristionJsonResult(accessTypeRoles);
+
+        AccessTypeRolesJurisdictionResults jurisdictionResults = new AccessTypeRolesJurisdictionResults();
+        jurisdictionResults.setJurisdictions(jurisdictions);
+        return  jurisdictionResults;
+    }
+
+    private List<AccessTypeRolesJurisdictionResult> buildJuristionJsonResult(
+        List<AccessTypeRolesField> accessTypeRoles) {
+        List<AccessTypeRolesJurisdictionResult>  jurisdictions = new ArrayList<AccessTypeRolesJurisdictionResult>();
```
Comment (abul-syed) [inline rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java:75]:
```suggestion
        List<AccessTypeRolesJurisdictionResult>  jurisdictions = new ArrayList<>();
```

File: rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+        role.setGroupRoleName(accessTypeRole.getGroupRoleName());
+        /***** Set with getIdOfCaseType Saved previously from casetypeId before it is copied and is = null when
+         * case is copied the id is null as (Property "id") has no write accessor
+         * ******/
+        role.setCaseTypeId(accessTypeRole.getIdOfCaseType().toString());
+        role.setOrganisationalRoleName(accessTypeRole.getOrganisationalRoleName());
+        role.setCaseGroupIdTemplate(accessTypeRole.getCaseAccessGroupIdTemplate());
+
+        List<AccessTypeRolesRoleResult> accessTypeRolesRoleResults = new ArrayList<AccessTypeRolesRoleResult>();
```
Comment (abul-syed) [inline rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java:117]:
```suggestion
        List<AccessTypeRolesRoleResult> accessTypeRolesRoleResults = new ArrayList<>();
```

**Review by abul-syed — CHANGES_REQUESTED** [review]
few changes

File: rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java (hunk 6 above)
Comment (patelila) [reply rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java:64]: image

if the body has nothing specified then the request will be a Bad Request (401).

File: repository/src/test/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesJurisdictionResultsTest.java [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+        accessTypeRolesResult.setRoles(accessTypeRolesRoleResults);
+
+        accessTypeRolesResults.add(accessTypeRolesResult);
+        accessTypeRolesJurisdictionResult.setAccessTypeRoles(accessTypeRolesResults);
+        accessTypeRolesJurisdictions.add(accessTypeRolesJurisdictionResult);
+        jurisdictionResults.setJurisdictions(accessTypeRolesJurisdictions);
+
+        //doReturn(controller.retrieveAccessTypeRoles(organisationProfileIds))
+        // .when(accessTypeRolesJurisdictions).get(0);
```
Comment (helenbirdhmcts) [inline repository/src/test/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesJurisdictionResultsTest.java:67]: commented out lines, should these be removed?

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesField.java [hunk 10]
```diff
@@ -28,5 +28,4 @@ public class AccessTypeRolesField {
     private String caseAssignedRoleField;
     private Boolean groupAccessEnabled;
     private String caseAccessGroupIdTemplate;
-
```
Comment (helenbirdhmcts) [inline repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesField.java:31]: Possibly revert this change

File: rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+        if (organisationProfileIds != null
+            && organisationProfileIds.getOrganisationProfileIds() != null
+            && !organisationProfileIds.getOrganisationProfileIds().isEmpty()) {
+            //Get only access_types with caseTypeId's by OrganisationProfileIds
+            List<AccessTypeRolesEntity> accessTypeRolesEntities = accessTypeRolesRepository
+                .findByOrganisationProfileIds(organisationProfileIds.getOrganisationProfileIds());
+            accessTypeRoles = accessTypeRolesEntities.stream().map(entityToResponseDTOMapper::map)
+                .collect(Collectors.toList());
+        } else {
+            // get all access Types  with caseTypeId's
```
Comment (helenbirdhmcts) [inline rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java:67]: do we need this comment the method findAllWithCaseTypeIds is descriptive enough I think?

File: rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java (hunk 6 above)
Comment (helenbirdhmcts) [reply rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java:73]: do we have a unit test that is testing this?

File: rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java [hunk 12]
```diff
@@ -0,0 +1,336 @@
...
+
+        assertFalse("accessTypeRolesJurisdictionResults is null or empty",
+            accessTypeRolesJurisdictionResults == null
+            || accessTypeRolesJurisdictionResults.getJurisdictions().isEmpty());
+
+        AccessTypeRolesJurisdictionResults finalAccessTypeRolesJurisdictionResults = accessTypeRolesJurisdictionResults;
+        assertAll(
+            () -> MatcherAssert.assertThat(finalAccessTypeRolesJurisdictionResults.getJurisdictions().size(),
+                is(4)),
+            //() -> assertThat(mvcResult.get(0).getOrganisationProfileId(), is(orgProfileIds.get(0)))
```
Comment (helenbirdhmcts) [inline rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java:194]: this assertThat is commented out, should it be removed?

File: rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java [hunk 13]
```diff
@@ -0,0 +1,336 @@
...
+    void shouldHandleEmptyOrganisationProfileIds() throws Exception {
+        OrganisationProfileIds organisationProfileIds = new OrganisationProfileIds();
+        //organisationProfileIds.setOrganisationProfileIds(orgProfileIds);
+
+        ObjectMapper objmapper = new ObjectMapper();
+        String response = objmapper.writeValueAsString(new AccessTypeRolesField());
+        String request = objmapper.writeValueAsString(organisationProfileIds);
+
+        final MvcResult mvcResult = mockMvc.perform(post("/api/retrieve-access-types")
+                //.contextPath("/api")
```
Comment (helenbirdhmcts) [inline rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java:217]: context path is commented out, should it be removed?

File: rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java [hunk 14]
```diff
@@ -0,0 +1,336 @@
...
+
+        assertFalse("accessTypeRolesJurisdictionResults is null or empty",
+            accessTypeRolesJurisdictionResults == null
+                || accessTypeRolesJurisdictionResults.getJurisdictions().isEmpty());
+
+        AccessTypeRolesJurisdictionResults finalAccessTypeRolesJurisdictionResults = accessTypeRolesJurisdictionResults;
+        assertAll(
+            () -> MatcherAssert.assertThat(finalAccessTypeRolesJurisdictionResults.getJurisdictions().size(),
+                is(4)),
+            //() -> assertThat(mvcResult.get(0).getOrganisationProfileId(), is(orgProfileIds.get(0)))
```
Comment (helenbirdhmcts) [inline rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java:238]: assertThat commented out, should it be removed?

File: rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java [hunk 15]
```diff
@@ -0,0 +1,331 @@
...
+        caseTypeEntity.setVersion(version.getNumber());
+        caseTypeEntity.setJurisdiction(jurisdiction);
+        caseTypeEntity.setName(caseTypeId);
+        return caseTypeEntity;
+    }
+
+    private void setupAccessTypeRolesEntity(AccessTypeRolesEntity accessTypeRolesEntity, String caseTypeId,
+                                            String organisationProfileId) {
+
+        //String caseTypeId = "get-test";
```
Comment (helenbirdhmcts) [inline rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java:262]: Comment above could be deleted?

File: rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java [hunk 16]
```diff
@@ -0,0 +1,331 @@
...
+        CaseTypeEntity caseTypeReturned = createCaseType(caseTypeId, caseTypeId);
+
+        //accessTypeRolesField = createAccessTypeRolesField(caseTypeId );
```
Comment (helenbirdhmcts) [inline rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java:265]: comment above could be deleted?

File: rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java [hunk 17]
```diff
@@ -0,0 +1,331 @@
...
+        accessTypeRolesEntity.setCaseAssignedRoleField("applicant1OrganisationPolicy");
+        accessTypeRolesEntity.setCaseAccessGroupIdTemplate("CIVIL:all:CIVIL:AS1:$ORGID$");
+
+        AccessTypeRolesField accessTypeRolesField;
+
+        accessTypeRolesField = entityToResponseDTOMapper.map(accessTypeRolesEntity);
+
+    }
+
+    void setUpAccessTypeRoleData() {
```
Comment (helenbirdhmcts) [inline rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java:291]: should this be 
private void setUpAccessTypeRoleData() {

File: rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+    @PostMapping(value = "/retrieve-access-types",consumes = {"application/json"}, produces = {"application/json"})
+    @ApiOperation(value = "Get all accessTypes for all caseTypes", response = AccessTypeRolesJurisdictionResult.class,
+        responseContainer = "List")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "List of valid access types request"),
+        @ApiResponse(code = 401, message = "Unauthorised request"),
+        @ApiResponse(code = 403, message = "Bad request")
+    })
+    public AccessTypeRolesJurisdictionResults retrieveAccessTypeRoles(
+        @Valid @RequestBody OrganisationProfileIds organisationProfileIds) {
```
Comment (abul-syed) [inline rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java:55]:
```suggestion
        @RequestBody(required=false) OrganisationProfileIds organisationProfileIds) {
```

to allow empty request body to be sent. also don't need `@Valid` - don't see any validation on the `OrganisationProfileIds`class

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/OrganisationProfileIds.java [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.util.List;
+
+@ApiModel(description = "")
+public class OrganisationProfileIds {
+
+    private List<String> organisationProfileIds = null;
+
+    @ApiModelProperty(value = "")
+    @JsonProperty("organisation_profile_ids")
```
Comment (abul-syed) [inline repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/OrganisationProfileIds.java:15]: should this be camel case?

**Review by abul-syed — COMMENTED** [review]
minor changes

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/OrganisationProfileIds.java (hunk 19 above)
Comment (patelila) [reply repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/OrganisationProfileIds.java:15]: I am not sure when this was created so I don't know if this is documented somewhere. I will have to ask Helen.

File: rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java [hunk 20]
```diff
@@ -0,0 +1,339 @@
...
+    private static final String JURISDICTION2 = "BEFTA_MASTER2";
+    @Mock
+    private AccessTypeRolesEntity accessTypeRolesEntity;
+    @Mock
+    private AccessTypeRolesEntity accessTypeRolesEntity1;
+    @Mock
+    private AccessTypeRolesEntity accessTypeRolesEntity2;
+    private AccessTypeRolesJurisdictionResults jurisdictionResults;
+    private List<AccessTypeRolesJurisdictionResult> accessTypeRolesJurisdictions;
+
```
Comment (helenbirdhmcts) [inline rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesControllerTest.java:79]: I think we need a test that verifies that the resulting JSON looks as its meant to.

File: rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java [hunk 21]
```diff
@@ -0,0 +1,147 @@
...
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "List of valid access types request"),
+        @ApiResponse(code = 401, message = "Unauthorised request"),
+        @ApiResponse(code = 403, message = "Bad request")
+    })
+    public AccessTypeRolesJurisdictionResults retrieveAccessTypeRoles(
+        @RequestBody(required = false) @Valid OrganisationProfileIds organisationProfileIds) {
+        List<AccessTypeRolesField> accessTypeRoles;
+
+        if (organisationProfileIds != null
```
Comment (abul-syed) [inline rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java:59]: maybe core business logic should live in a service class, e.g. `AccessTypeRolesService`?

**Review by abul-syed — COMMENTED** [review]
have just left 1 comment. other than that lgtm

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/entity/CaseTypeEntity.java [hunk 22]
```diff
@@ -43,7 +44,9 @@
 )
 public class CaseTypeEntity implements Serializable, Versionable {
 
-    private static final long serialVersionUID = 542723327314434924L;
```
Comment (sanjayparekh-hmcts) [inline repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/entity/CaseTypeEntity.java:46]: why did we need to change the UID here?

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesJurisdictionResults.java [hunk 23]
```diff
@@ -0,0 +1,24 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+@ApiModel(description = "jurisdictions")
+public class AccessTypeRolesJurisdictionResults {
+
+    private List<AccessTypeRolesJurisdictionResult> jurisdictions = new ArrayList<>();
+
+    @ApiModelProperty(value = "")
```
Comment (sanjayparekh-hmcts) [inline repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesJurisdictionResults.java:15]: why id this here  blank?

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesResult.java [hunk 24]
```diff
@@ -0,0 +1,111 @@
...
+    @JsonProperty("organisationProfileId")
+    public String getOrganisationProfileId() {
+        return organisationProfileId;
+    }
+
+    public void setOrganisationProfileId(String organisationProfileId) {
+        this.organisationProfileId = organisationProfileId;
+    }
+
+    @ApiModelProperty(required = true, value = "")
```
Comment (sanjayparekh-hmcts) [inline repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/AccessTypeRolesResult.java:32]: not sure about the any advantage f having these ApiModelProperty with blank values - does it need any description?

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java [hunk 25]
```diff
@@ -326,6 +324,10 @@ static List<CaseEventFieldComplex> map(List<? extends EventComplexTypeEntity> ev
     @Mapping(source = "caseType.reference", target = "caseTypeId")
     Category map(CategoryEntity categoryEntity);
 
+    @Mapping(source = "caseTypeId.reference", target = "caseTypeId.reference")
+    @Mapping(source = "caseTypeId", target = "caseTypeId")
```
Comment (sanjayparekh-hmcts) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java:328]: is this needed when the source and target are the same?

File: rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java (hunk 21 above)
Comment (sanjayparekh-hmcts) [reply rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java:59]: definitely need to move this out of the controller into a service

**Review by sanjayparekh-hmcts — CHANGES_REQUESTED** [review]

File: rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java [hunk 26]
```diff
@@ -0,0 +1,51 @@
...
+import uk.gov.hmcts.ccd.definition.store.repository.model.OrganisationProfileIds;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesJurisdictionResult;
+import uk.gov.hmcts.ccd.definition.store.repository.model.AccessTypeRolesJurisdictionResults;
+import uk.gov.hmcts.ccd.definition.store.rest.service.AccessTypeRolesService;
+
+import javax.validation.Valid;
+
+@RestController
+@Api(value = "/api")
+@RequestMapping(value = "/api")
```
Comment (sanjayparekh-hmcts) [inline rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java:24]: minor issue - I dont really know why we added /api as root here, but if ccd are OK with that, then its fine

**Review by sanjayparekh-hmcts — APPROVED** [review]

File: rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java (hunk 26 above)
Comment (RebeccaBaker) [reply rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/AccessTypeRolesController.java:24]: Can this be removed? as it doesn't match the ticket description or any corresponding documentation you may have

**Review by RebeccaBaker — APPROVED** [review]

**Review by patelila — APPROVED** [review]
I have tested and reviewed the code changes that Helen has made to fix the serialisation error. The GA-20 works fine. Was able to create case with no serialisation errors.
